distributed/shuffle/_worker_plugin.py

```diff
@@ -1005,8 +1015,9 @@ def convert_chunk(data: bytes) -> np.ndarray:
     shards: dict[NDIndex, np.ndarray] = {}
 
     while file.tell() < len(data):
-        index, shard = pickle.load(file)
-        shards[index] = shard
+        for tpl in pickle.load(file):
+            index, shard = tpl
```

Comment:
```suggestion
        for index, shard in pickle.load(file):
```

Reply:
Previously, each `pickle.load` produced a single `index, shard` pair (and there's some precondition that these indices don't overlap). Now each `pickle.load` produces multiple `index, shard` pairs. Is there still the same implicit precondition that every `index` must be unique?

Reply:
Yes. Essentially, we just batch a bunch of `index, shard` pairs that would have required multiple `pickle.load` steps previously.